src/agent/secrets.clj

```diff
@@ -74,6 +75,41 @@
           [nil (format "Secret '%s' not found" secret-path)]))))
 
 
+;; filesystem
+(declare get-filesystem-secret)
+
+(def default-filesystem-location
+  "secrets.json")
+
+(defn persist-secret [secret]
+  (log/info "Persisting filesystem secrets")
+  (try
+    (spit default-filesystem-location secret)
+    [secret nil]
+    (catch Exception e
+      (log/error (format "Failed to persist filesystem secrets with error: %s" e))
+      [nil "failed to persist filesystem secrets"])))
+
+(defmethod fetch "filesystem" [task]
+  (log/info (format "Fetching secrets from filesystem for task id: %s" (:id task)))
+  (err/err->> task
+              validate-secret-path
+              get-filesystem-secret
+              (fn[result] [(assoc result :secrets ((keyword (:secret-path task)) (:secrets result)))])))
+
+(defn get-filesystem-secret [task]
+  (try
+    (if-let [content (slurp default-filesystem-location)]
+      (err/err->> content
+                  json->map
+                  (fn [secrets] [(assoc task :secrets secrets) nil]))
+      (do (log/error (format "failed to get filesystem '%s'" (:secret-path task)))
+          [nil (format "Failed to read filesystem secret '%s'" (:secret-path task))]))
+    (catch Exception e
+      (log/error (format "failed to get filesystem '%s'" (:secret-path task)))
```

Comment:
It's missing the exception error, e.g.:
```clj
(log/error e "<ERROR MSG>")
```

Add a sentry line to report to it:

```clj
(sentry-task-logger e task "failed to retrieving secret from filesystem")
```

Reply:
I added the sentry line.

However, for the error message, I kept the same app pattern for other secrets error (to print the error as a formatted string).
```
(log/error (format "failed to get filesystem '%s' with error %s" (:secret-path task) e))
```